sardana_alba/ctrl/LimaRoICoTiCtrl.py

```diff
@@ -147,23 +147,26 @@ def StateAll(self):
             return
 
         self._last_image_ready = self._limaroi.read_attribute(attr).value
-        if (self._last_image_ready < (self._repetitions - 1) and
-                self._last_image_ready != -2):
+        
+        if self._last_image_ready == -2:
+            self._state = State.On
+            self._status = 'Initialized'
+        elif self._last_image_ready < (self._repetitions - 1):
             self._state = State.Moving
             self._status = 'Taking data'
-        else:
+            self._read = False
+        elif self._read:
             self._state = State.On
-            # self._clean_acquisition()
-            if self._last_image_ready == -2:
-                self._status = "Not images in buffer"
-            else:
-                self._status = "RoI computed"
+            self._status = "RoI computed"
+        else:
```

Comment:
Not setting a `self._state` here can lead to errors when querying the state if for example we restart the controller and the acquisition has finished